tools/security.js

```
@@ -56,6 +56,25 @@ const { client } = require('./database/redis');
 const { loggerAuth } = require(`${SERVER_PATH}/config/logger`);
 const moment = require('moment');
 const { generateHash } = require(`${SERVER_PATH}/utils/security`);
+const geoip = require('geoip-lite');
+
+const getCountryFromIp = (ip) => {
+	const geo = geoip.lookup(ip);
+	if (!geo) {
+		return '';
+	}
+	return geo.country;
+};
+
+const checkIp = async (remoteip = '') => {
+	const dataGeofence = getKitConfig().black_list_countries;
+	if (dataGeofence && dataGeofence.length > 0 && remoteip) {
+		if (dataGeofence.includes(getCountryFromIp(remoteip))) {
+			return reject(new Error('ERROR IP LOCATION'));
```

@hoan-bitholla You can just throw an error here since this is an async function